(Combination) of configuration options does not work

We are trying out visual configurations of PSPDFKit (for Ionic/Angular/Cordova) and on the custom configuration options suggested [here](https://github.com/PSPDFKit/PSPDFKit-Cordova/tree/master/docs/android#options) we are getting runtime errors upon initializing the plugin, telling that a particular value is not valid (thus needed to be commented out or changed to default value). For us it worked only after initializing with empty options or reducing the number of options to what you see here (what is not commented out):

```typescript
    const options: PSPDFKitOptions = {
        backgroundColor: "#EFEFEF", // hex-color of the page background
        // disableAnnotationList: false, // hide annotation list (default: false)
        disableAnnotationNoteHinting: false, // hide small notes next to annotations that have a text set (default: false)
        // disableBookmarkEditing: false, // hide bookmark editing (default: false)
        // disableBookmarkList: false, // hide bookmark list (default: false)
        disableCopyPaste: false, // disable annotation copy/paste (default: false)
        // disableDocumentEditor: false, // hide document editor (default: false)
        // disableOutline: false, // hide the outline menu (default: false)
        // disablePrinting: false, // hide option to print (default: false)
        disableUndoRedo: false, // disable undo/redo system (default: false)
        // hidePageLabels: false, // hide page labels (if available in PDF) in page overlay and thumbnail grid (default: false)
        // // hidePageNumberOverlay: false, // hide the overlay showing the current page (default: false)
        // // hideSettingsMenu: false, // hide the settings menu (default: false)
        // // thumbnailBarMode: PSPDFKit.ThumbnailBarMode.THUMBNAIL_BAR_MODE_DEFAULT, // show static thumbnail bar. Also valid: THUMBNAIL_BAR_MODE_DEFAULT, THUMBNAIL_BAR_MODE_SCROLLABLE
        // // hideThumbnailGrid: false, // hide the thumbnail grid menu (default: false)
        pageFitMode: PSPDFKit.PageFitMode.FIT_TO_WIDTH, // also valid: PSPDFKit.PageFitMode.FIT_TO_SCREEN
        // scrollDirection: PSPDFKit.PageScrollDirection.VERTICAL, // also valid: PSPDFKit.PageScrollDirection.HORIZONTAL
        // scrollMode: PSPDFKit.ScrollMode.CONTINUOUS, // also valid: PSPDFKit.ScrollMode.PER_PAGE
        // // Configures which share actions should be visible in the user interface. (default: all enabled)
        shareFeatures: [
            /** Document sharing inside the activity. */
            PSPDFKit.ShareFeatures.DOCUMENT_SHARING,
            /** Sharing of embedded files (on file annotations). */
            PSPDFKit.ShareFeatures.EMBEDDED_FILE_SHARING,
            /** Sharing of text from selected free text annotations. */
            PSPDFKit.ShareFeatures.FREE_TEXT_ANNOTATION_SHARING,
            /** Sharing of selected image annotations. */
            PSPDFKit.ShareFeatures.IMAGE_SHARING,
            /** Sharing of text from selected note annotations. */
            PSPDFKit.ShareFeatures.NOTE_ANNOTATION_SHARING,
            /** Sharing of text from annotation contents or comments. */
            PSPDFKit.ShareFeatures.NOTE_EDITOR_CONTENT_SHARING,
            /** Sharing of selected text. */
            PSPDFKit.ShareFeatures.TEXT_SELECTION_SHARING
        ],
        invertColors: false, // invert rendered colors (default: false)
        toGrayscale: false, // render document in grayscale only (default: false)
        // startZoomScale: 2.0, // initial zoom value (default: 1.0)
        // maxZoomScale: 10.0, // maximum zoom factor when zooming into a page (default: 15.0)
        zoomOutBounce: true, // "bounce" animation when pinch-zooming out (default: true)
        // page: 2, // initial page number (default: 0, i.e. the first page)
        useImmersiveMode: false, // activate Android's immersive app mode (default: false)
        // disableSearch: false, // completely deactivate document search (default: false)
        // searchType: PSPDFKit.SearchType.SEARCH_MODULAR, // also valid: PSPDFKit.SearchType.SEARCH_INLINE
        // autosaveEnabled: true, // automatically save document changes on exit (default: true)
        // annotationEditing: {
        //     enabled: true, // activate annotation editing (default: true)
        //     creatorName: "John Doe" // author name written into new annotations (default: null)
        // }
    };
        PSPDFKit.present(path, options, () => {}, () => {});
```
Do the options work only in certain combinations or are they deprecated and the docs need to be updated?

We are using `"pspdfkit-cordova": "1.0.9"` with ionic 4 and cordova 9

This is on Android. For example, if I uncomment hidePageLabels in the code above, I get:

`E/PluginManager: Uncaught exception from plugin
    com.pspdfkit.cordova.PSPDFKitPluginException: Error while parsing option 'hidePageLabels'
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.parseOptionsToConfiguration(ShowDocumentAction.java:187)
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.execAction(ShowDocumentAction.java:197)
        at com.pspdfkit.cordova.action.BasicAction.exec(BasicAction.java:41)
        at com.pspdfkit.cordova.action.ActionManager.executeAction(ActionManager.java:29)
        at com.pspdfkit.cordova.PSPDFKitPlugin.execute(PSPDFKitPlugin.java:128)
        at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98)
        at org.apache.cordova.PluginManager.exec(PluginManager.java:132)
        at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59)
        at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:336)
        at android.os.Looper.loop(Looper.java:174)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: java.lang.IllegalArgumentException: Invalid plugin option 'hidePageLabels'
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.parseOptionsToConfiguration(ShowDocumentAction.java:183)
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.execAction(ShowDocumentAction.java:197) 
        at com.pspdfkit.cordova.action.BasicAction.exec(BasicAction.java:41) 
        at com.pspdfkit.cordova.action.ActionManager.executeAction(ActionManager.java:29) 
        at com.pspdfkit.cordova.PSPDFKitPlugin.execute(PSPDFKitPlugin.java:128) 
        at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98) 
        at org.apache.cordova.PluginManager.exec(PluginManager.java:132) 
        at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59) 
        at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41) 
        at android.os.MessageQueue.nativePollOnce(Native Method) 
        at android.os.MessageQueue.next(MessageQueue.java:336) 
        at android.os.Looper.loop(Looper.java:174) 
        at android.os.HandlerThread.run(HandlerThread.java:67)
D/SystemWebChromeClient: http://localhost/plugins/pspdfkit-cordova/www/PSPDFKit.js: Line 1215 : Error while parsing option 'hidePageLabels'
I/chromium: [INFO:CONSOLE(1215)] "Error while parsing option 'hidePageLabels'", source: http://localhost/plugins/pspdfkit-cordova/www/PSPDFKit.js (1215)
`

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "pspdfkit-cordova",
-  "version": "1.0.10",
+  "version": "1.0.11",
   "description": "PSPDFKit Cordova Plugin for Android and iOS",
   "cordova": {
     "id": "pspdfkit-cordova",

--- a/plugin.xml
+++ b/plugin.xml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<plugin xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android" id="pspdfkit-cordova" version="1.0.10">
+<plugin xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android" id="pspdfkit-cordova" version="1.0.11">
   <engines>
     <engine name="cordova" version="&gt;=6.3.1" />
   </engines>

--- a/src/android/java/com/pspdfkit/cordova/action/document/ShowDocumentAction.java
+++ b/src/android/java/com/pspdfkit/cordova/action/document/ShowDocumentAction.java
@@ -96,7 +96,7 @@ public class ShowDocumentAction extends BasicAction {
         } else if ("backgroundColor".equals(option)) {
           builder.backgroundColor(Color.parseColor((String) value));
         } else if ("disableAnnotationList".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disableAnnotationList();
           } else {
             builder.enableAnnotationList();
@@ -104,13 +104,13 @@ public class ShowDocumentAction extends BasicAction {
         } else if ("disableAnnotationNoteHinting".equals(option)) {
           builder.setAnnotationNoteHintingEnabled(!(Boolean) value);
         } else if ("disableBookmarkEditing".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disableBookmarkEditing();
           } else {
             builder.enableBookmarkEditing();
           }
         } else if ("disableBookmarkList".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disableBookmarkList();
           } else {
             builder.enableBookmarkList();
@@ -122,25 +122,25 @@ public class ShowDocumentAction extends BasicAction {
             builder.enableCopyPaste();
           }
         } else if ("disableDocumentEditor".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disableDocumentEditor();
           } else {
             builder.enableDocumentEditor();
           }
         } else if ("disableOutline".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disableOutline();
           } else {
             builder.enableOutline();
           }
         } else if ("disablePrinting".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disablePrinting();
           } else {
             builder.enablePrinting();
           }
         } else if ("disableSearch".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.disableSearch();
           } else {
             builder.enableSearch();
@@ -150,19 +150,19 @@ public class ShowDocumentAction extends BasicAction {
         } else if ("disableUndoRedo".equals(option)) {
           builder.undoEnabled(!(Boolean) value);
         } else if ("hidePageLabels".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.hidePageLabels();
           } else {
             builder.showPageLabels();
           }
         } else if ("hidePageNumberOverlay".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.hidePageNumberOverlay();
           } else {
             builder.showPageNumberOverlay();
           }
         } else if ("hideSettingsMenu".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.hideSettingsMenu();
           } else {
             builder.showSettingsMenu();
@@ -171,7 +171,7 @@ public class ShowDocumentAction extends BasicAction {
           final CordovaThumbnailBarMode thumbnailBarMode = CordovaThumbnailBarMode.valueOf((String) value);
           builder.setThumbnailBarMode(thumbnailBarMode.androidThumbnailBarMode);
         } else if ("hideThumbnailGrid".equals(option)) {
-          if (((Boolean) value)) {
+          if ((Boolean) value) {
             builder.hideThumbnailGrid();
           } else {
             builder.showThumbnailGrid();

--- a/src/android/java/com/pspdfkit/cordova/action/document/ShowDocumentAction.java
+++ b/src/android/java/com/pspdfkit/cordova/action/document/ShowDocumentAction.java
@@ -95,39 +95,87 @@ public class ShowDocumentAction extends BasicAction {
           builder.autosaveEnabled((Boolean) value);
         } else if ("backgroundColor".equals(option)) {
           builder.backgroundColor(Color.parseColor((String) value));
-        } else if ("disableAnnotationList".equals(option) && ((Boolean) value)) {
-          builder.disableAnnotationList();
+        } else if ("disableAnnotationList".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disableAnnotationList();
+          } else {
+            builder.enableAnnotationList();
+          }
         } else if ("disableAnnotationNoteHinting".equals(option)) {
           builder.setAnnotationNoteHintingEnabled(!(Boolean) value);
-        } else if ("disableBookmarkEditing".equals(option) && ((Boolean) value)) {
-          builder.disableBookmarkEditing();
-        } else if ("disableBookmarkList".equals(option) && ((Boolean) value)) {
-          builder.disableBookmarkList();
+        } else if ("disableBookmarkEditing".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disableBookmarkEditing();
+          } else {
+            builder.enableBookmarkEditing();
+          }
+        } else if ("disableBookmarkList".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disableBookmarkList();
+          } else {
+            builder.enableBookmarkList();
+          }
         } else if ("disableCopyPaste".equals(option)) {
-          if ((Boolean) value) builder.disableCopyPaste();
-          else builder.enableCopyPaste();
-        } else if ("disableDocumentEditor".equals(option) && ((Boolean) value)) {
-          builder.disableDocumentEditor();
-        } else if ("disableOutline".equals(option) && ((Boolean) value)) {
-          builder.disableOutline();
-        } else if ("disablePrinting".equals(option) && ((Boolean) value)) {
-          builder.disablePrinting();
-        } else if ("disableSearch".equals(option) && ((Boolean) value)) {
-          builder.disableSearch();
+          if ((Boolean) value) {
+            builder.disableCopyPaste();
+          } else {
+            builder.enableCopyPaste();
+          }
+        } else if ("disableDocumentEditor".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disableDocumentEditor();
+          } else {
+            builder.enableDocumentEditor();
+          }
+        } else if ("disableOutline".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disableOutline();
+          } else {
+            builder.enableOutline();
+          }
+        } else if ("disablePrinting".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disablePrinting();
+          } else {
+            builder.enablePrinting();
+          }
+        } else if ("disableSearch".equals(option)) {
+          if (((Boolean) value)) {
+            builder.disableSearch();
+          } else {
+            builder.enableSearch();
+          }
         } else if ("shareFeatures".equals(option)) {
           builder.setEnabledShareFeatures(parseShareFeatures((JSONArray) value));
         } else if ("disableUndoRedo".equals(option)) {
           builder.undoEnabled(!(Boolean) value);
-        } else if ("hidePageLabels".equals(option) && ((Boolean) value)) {
-          builder.hidePageLabels();
-        } else if ("hidePageNumberOverlay".equals(option) && ((Boolean) value)) {
-          builder.hidePageNumberOverlay();
-        } else if ("hideSettingsMenu".equals(option) && ((Boolean) value)) {
-          builder.hideSettingsMenu();
+        } else if ("hidePageLabels".equals(option)) {
+          if (((Boolean) value)) {
+            builder.hidePageLabels();
+          } else {
+            builder.showPageLabels();
+          }
+        } else if ("hidePageNumberOverlay".equals(option)) {
+          if (((Boolean) value)) {
+            builder.hidePageNumberOverlay();
+          } else {
+            builder.showPageNumberOverlay();
+          }
+        } else if ("hideSettingsMenu".equals(option)) {
+          if (((Boolean) value)) {
+            builder.hideSettingsMenu();
+          } else {
+            builder.showSettingsMenu();
+          }
         } else if ("thumbnailBarMode".equals(option)) {
-          builder.setThumbnailBarMode(ThumbnailBarMode.valueOf((String) value));
-        } else if ("hideThumbnailGrid".equals(option) && ((Boolean) value)) {
-          builder.hideThumbnailGrid();
+          final CordovaThumbnailBarMode thumbnailBarMode = CordovaThumbnailBarMode.valueOf((String) value);
+          builder.setThumbnailBarMode(thumbnailBarMode.androidThumbnailBarMode);
+        } else if ("hideThumbnailGrid".equals(option)) {
+          if (((Boolean) value)) {
+            builder.hideThumbnailGrid();
+          } else {
+            builder.showThumbnailGrid();
+          }
         } else if ("memoryCacheSize".equals(option)) {
           builder.memoryCacheSize((Integer) value);
         } else if ("pageFitMode".equals(option)) {
@@ -211,13 +259,31 @@ public class ShowDocumentAction extends BasicAction {
         ImageDocumentUtils.isImageUri(context, uri)
             ? PdfActivityIntentBuilder.fromImageUri(context, uri)
             : PdfActivityIntentBuilder.fromUri(context, uri)
-                // Only set passwords for PDF documents, since image documents don't support passwords.
-                .passwords(password);
+            // Only set passwords for PDF documents, since image documents don't support passwords.
+            .passwords(password);
 
     final Intent launchIntent = builder.activityClass(CordovaPdfActivity.class)
         .configuration(configuration)
         .build();
     plugin.cordova.startActivityForResult(getPlugin(), launchIntent, 0);
     callbackContext.success();
+  }
+
+  /**
+   * Cordova representation of the PSPDFKit thumbnail bar mode, that maps to the Android specific
+   * thumbnail bar mode. This is here, to map the cross-platform API we provide in JavaScript to the
+   * Android API. Values in this enum match the {@code ThumbnailBarMode} that is defined inside the
+   * {@code PSPDFKit.js} file.
+   */
+  private enum CordovaThumbnailBarMode {
+    THUMBNAIL_BAR_MODE_DEFAULT(ThumbnailBarMode.THUMBNAIL_BAR_MODE_FLOATING),
+    THUMBNAIL_BAR_MODE_SCROLLABLE(ThumbnailBarMode.THUMBNAIL_BAR_MODE_SCROLLABLE),
+    THUMBNAIL_BAR_MODE_NONE(ThumbnailBarMode.THUMBNAIL_BAR_MODE_NONE);
+
+    public final ThumbnailBarMode androidThumbnailBarMode;
+
+    CordovaThumbnailBarMode(ThumbnailBarMode androidThumbnailBarMode) {
+      this.androidThumbnailBarMode = androidThumbnailBarMode;
+    }
   }
 }

